lib/beaker-pe/install/pe_utils.rb

```diff
@@ -535,6 +505,82 @@ def do_install hosts, opts = {}
+        #
+        # @note that these steps aren't necessary for all hosts. Specifically,
+        #   'agent_only' hosts do not require these steps to be executed.
+        #
+        # @param [Array<Host>] hosts Hosts to prepare
+        # @param [Hash{Symbol=>String}] local_options Local options, used to
+        #   pass misc configuration required for the prep steps
+        #
+        # @return nil
+        def prepare_hosts(hosts, local_options={})
+          use_all_tar = ENV['PE_USE_ALL_TAR'] == 'true'
+          hosts.each do |host|
+            host['pe_installer'] ||= 'puppet-enterprise-installer'
+            if host['platform'] !~ /windows|osx/
+              platform = use_all_tar ? 'all' : host['platform']
+              version = host['pe_ver'] || local_options[:pe_ver]
+              host['dist'] = "puppet-enterprise-#{version}-#{platform}"
+            elsif host['platform'] =~ /osx/
+              version = host['pe_ver'] || local_options[:pe_ver]
+              host['dist'] = "puppet-enterprise-#{version}-#{host['platform']}"
+            elsif host['platform'] =~ /windows/
+              version = host[:pe_ver] || local_options['pe_ver_win']
+              is_config_32 = true == (host['ruby_arch'] == 'x86') || host['install_32'] || local_options['install_32']
+              should_install_64bit = !(version_is_less(version, '3.4')) && host.is_x86_64? && !is_config_32
+              #only install 64bit builds if
+              # - we are on pe version 3.4+
+              # - we do not have install_32 set on host
+              # - we do not have install_32 set globally
+              if !(version_is_less(version, '3.99'))
+                if should_install_64bit
+                  host['dist'] = "puppet-agent-#{version}-x64"
+                else
+                  host['dist'] = "puppet-agent-#{version}-x86"
+                end
+              elsif should_install_64bit
+                host['dist'] = "puppet-enterprise-#{version}-x64"
+              else
+                host['dist'] = "puppet-enterprise-#{version}"
+              end
+            end
+            host['working_dir'] = host.tmpdir(Time.new.strftime("%Y-%m-%d_%H.%M.%S"))
+          end
+        end
+
+        # Gets the puppet-agent version, hopefully from the host or local options.
+        # Will fall back to reading the `aio_agent_build` property on the master
+        # if neither of those two options are passed
+        #
+        # @note This method does have a side-effect: if it reads the
+        #   `aio_agent_build` property from master, it will store it in the local
+        #   options hash so that it won't have to do this more than once.
+        #
+        # @param [Beaker::Host] host Host to get puppet-agent for
+        # @param [Hash{Symbol=>String}] local_options local method options hash
+        #
+        # @return [String] puppet-agent version to install
+        def get_puppet_agent_version(host, local_options={})
```

Comment:
@tvpartytonight for your first point, I think we're somewhat inconsistent for sure about the terms. I think it's fair to say that `puppet_agent` is a container term that would hold all instances of `aio` as well as agents outside of `aio`, so it's used when we want to be a little more general than that. But we haven't been 100% about that, because the installers that were added for `aio` when that was going down were all called `puppet_agent` as well. Since we're coming closer and closer to EOL'ing the 3.8 series from before `aio` came out that this distinction is becoming less and less meaningful, however, so I'm fine just letting it fall away naturally. But it would be good to have a more explicit label on these things so I'm open to moving. Unfortunately the challenge to that has always been changing signatures and hence the API, but perhaps we can get around that with other methods.

For your second comment, I've done a little looking, and commented on this asking @jpartlow for more info [in the ticket](https://tickets.puppetlabs.com/browse/BKR-831?focusedCommentId=336387&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-336387). Hopefully we can resolve that there, and figure out what the right move is for this code.